client/src/routes/settings/+page.svelte

```diff
@@ -1,31 +1,212 @@
 <script lang="ts">
-  import { deleteAccount, logOut } from "$lib/firebase/auth";
+  import { deleteAccount, linkWithGoogle, linkWithMicrosoft, linkWithPassword, logOut } from "$lib/firebase/auth";
   import { goto } from "$app/navigation";
   import Dialog, { Title, Content, Actions } from "@smui/dialog";
   import Button, { Label } from "@smui/button";
+  import { Svg, Icon } from "@smui/common";
+  import IconButton from "@smui/icon-button";
+  import Textfield from "@smui/textfield";
+  import { authStore as user } from "$stores/auth";
 
   let showDeletionPrompt = false;
-  let errorMsg = "";
+  let showOptions = false;
+  let showPassword = false;
   let errPrompt = false;
+  let googleLinked = false;
+  let microsoftLinked = false;
+  let linkPass = false;
+  let passCheck = false;
+
+  let errorMsg = "";
+  let password = "";
+  let confirmPassword = "";
+  let googleErr = "";
+  let microsoftErr = "";
+  let passErr = "";
 
+  function outputErrMsg() {
```

Comment:
I would prefer if this function had an argument and a return instead of mutating the state